src/quacc/calculators/espresso/espresso.py

```diff
@@ -80,17 +81,29 @@ def write_input(
         -------
         None
         """
+
         directory = Path(directory)
         self._outdir_handler(parameters, directory)
 
-        write(
-            directory / self.inputname,
-            atoms,
-            binary=self.binary,
-            properties=properties,
-            pseudo_dir=str(profile.pseudo_path),
-            **parameters,
-        )
+        fd = Path.open(directory / self.inputname, "w")
+
+        if self.binary == "pw":
+            write(
+                filename=directory / self.inputname,
+                images=atoms,
+                format="espresso-in",
+                pseudo_dir=str(profile.pseudo_path),
+                properties=properties,
+                **parameters,
+            )
+        elif self.binary == "ph":
+            write_espresso_ph(fd=fd, properties=properties, **parameters)
+        else:
+            write_fortran_namelist(
+                fd, binary=self.binary, properties=properties, **parameters
+            )
+
+        fd.close()
```

Comment:
Can probably do `with Path.open(...)` so you don't have to explicitly `fd.close()`.